src/main/java/mate/academy/EventManager.java

```diff
@@ -1,19 +1,38 @@
 package mate.academy;
 
+import java.util.List;
+import java.util.concurrent.CopyOnWriteArrayList;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeUnit;
+
 public class EventManager {
-    public void registerListener(EventListener listener) {
+    private final List<EventListener> eventListeners = new CopyOnWriteArrayList<>();
+    private final ExecutorService executorService = Executors.newCachedThreadPool();
```

Comment:
Using Executors.newCachedThreadPool() can potentially create a large number of threads if the application receives a high volume of events. This may lead to resource exhaustion. Consider setting a limit on the number of threads or using a different type of thread pool.